packages/shopping/src/services/email.service.ts

```diff
@@ -0,0 +1,41 @@
+import {bind, BindingScope} from '@loopback/core';
+import {EmailTemplate, NodeMailer, User} from '../models';
+import {createTransport} from 'nodemailer';
+
+@bind({scope: BindingScope.TRANSIENT})
+export class EmailService {
+  private static async setupTransporter() {
+    return createTransport({
+      host: process.env.SMTP_SERVER,
+      port: +(process.env.SMTP_PORT ?? 0),
```

Comment:
Note sure, might give NaN in some circumstances.  `+process.env.SMTP_PORT || 0`, I think would better?  If it tests OK, though... 👍

Reply:
NaN is unavoidable since entering `SMTP_PORT=gccg` would result in NaN either with Number() or otherwise. I reused existing logic for PORT since its familiar at this point. `+process.env.SMTP_PORT ` wont work since the value is possibly undefined

Reply:
wouldnt it be better to have `587` as the default value?

Reply:
to be honest any value here should be fine. When i was testing for example I was using my own SMTP server running on port 25 and Gmail runs on port 587. In any case there will be a specific error for wrong port so 0/587 is still fine to demonstrate that port is a number number value

Reply:
Avoidable NaN: [This is the way](https://www.typescriptlang.org/play?#code/MYewdgzgLgBNBOBLMBzAcgQzTAvDA5AEYZgkb4DcAsAFCiSwLLoCuAtrgQKyW233QY8AKYRM2PAGoAFE1TiYAfkUwADAEpqdcBBAAbYQDo9IFNIAGMueixKVGmAC4YAEgDeIsVgC+5zfx1YTzR2TisoJHlQ5TV-bUh9IxMzS1kI5hCOGIdnd2D2Xzi+eMFgrCgAC0QIAEkIABUK4QB1DABPMOsFAB9utS0BRONTC0ku2161J1cPUXFK6rrGlvbCgcChOfYF2oam1o6pLtDJ1XWEg2GUsfSojlPpvK22HaX91b8KIA)

Reply:
That example is different to our use case `process.env.SMTP_PORT` is not always a string but rather undefined hence `+undefined || 0` would always result in linting errors. I would really prefer to reuse existing implementation on this 1

Reply:
I was trying to accommodate your preference for the `+` operator, so the original suggestion was the most suitable? `Number(process.env.SMTP_PORT) || 0`  As a bonus, the clarity would be self documenting without the need to know about the `+` trick.

Reply:
Personally, to me the suitable suggestion would be to reuse the existing pattern

Reply:
@dougal83 resolving this one. Please open an issue or a PR to close off on this discussion.